geode-management/src/main/java/org/apache/geode/management/api/Groupable.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.management.api;
+
+import javax.xml.bind.annotation.XmlTransient;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.apache.commons.lang3.StringUtils;
+
+import org.apache.geode.annotations.Experimental;
+
+/**
+ * a thing that can have a group. if it has no group, then it belongs to the default group CLUSTER
+ * See also MultiGroupCacheElement for things that can have more than one group
+ */
+@Experimental
+public interface Groupable {
```

Comment:
An idea we could iron out: a better name for this interface.

The intention behind this interface (along with `RestfulEndpoint` and `CMSJsonSerializable` and `MultiGroupCacheElement`) is that they are features/qualities of request/response objects from the v2 api endpoint. Is there a consistent naming convention between these interfaces that make it clear how they should be used?

Reply:
I'm fine with renaming them however you want.  Want to prefix `CMS` on everything for consistency?  I agree some better javadoc comments on how the pieces fit together are needed, maybe even a wiki tutorial covering the  'How To Add A Thing to the ConfigurationManagementService'?  I'll quickly summarize here: every CacheElement is Groupable (has a group) because that's how configuration is organized.  Most CMS calls operate on something that extends CacheElement, so the core of CMS would up with some code that assumed everything belonged to a group.  To break away from requiring everything to be a CacheElement, I conditionalized those parts of the core code that expect things to belong to groups, to only execute those portions of the code if Groupable is implemented.  This is definitely not the most elegant solution, but we did try once before to use subclassing to factor out the "non-groupy" code from the "groupy" code and it turned out to be even less pretty than this approach.

Others: CMSJsonSerializable is already clearly documented (I think) and if you fail to use it when adding a new service, the compiler will let you know.  RestfulEndpoint is pre-existing (and not significantly changed by this PR other that to add better documentation).  MultiGroupCacheElement is also pre-existing, but solves a similar problem as Groupable in the sense that it conditionalizes core CMS logic to differentiate between things that can only belong to a single group vs things that can belong to more than one group.

Reply:
If you think something needs to be prefixed with a common name then it is a package name. Prefixing class names isn't very readable and redundant to what packages are supposed to be.